internal/pkg/cli/env_init.go

```diff
@@ -69,53 +72,100 @@ func (opts *InitEnvOpts) Validate() error {
 	}
 
 	deployEnvInput := &deploy.CreateEnvironmentInput{
 		Name:                     opts.EnvName,
-		Project:                  viper.GetString(projectFlag),
+		Project:                  opts.projectName,
 		Prod:                     opts.IsProduction,
 		PublicLoadBalancer:       true, // TODO: configure this based on user input or application Type needs?
-		ToolsAccountPrincipalARN: identity.ARN,
+		ToolsAccountPrincipalARN: caller.ARN,
 	}
 
-	opts.prog.Start("Preparing deployment...")
+	opts.prog.Start(fmt.Sprintf("Proposing infrastructure changes for the %s environment", color.HighlightUserInput(opts.EnvName)))
 	if err := opts.envDeployer.DeployEnvironment(deployEnvInput); err != nil {
 		var existsErr *cloudformation.ErrStackAlreadyExists
 		if errors.As(err, &existsErr) {
 			// Do nothing if the stack already exists.
-			opts.prog.Stop("Done!")
+			opts.prog.Stop("")
 			log.Successf("Environment %s already exists under project %s! Do nothing.\n",
-				color.HighlightUserInput(opts.EnvName), color.HighlightResource(viper.GetString(projectFlag)))
+				color.HighlightUserInput(opts.EnvName), color.HighlightResource(opts.projectName))
 			return nil
 		}
-		opts.prog.Stop("Error!")
+		opts.prog.Stop(fmt.Sprintf("%s Failed to accept changes for the %s environment", color.ErrorMarker, color.HighlightUserInput(opts.EnvName)))
 		return err
 	}
-	opts.prog.Stop("Done!")
-	opts.prog.Start("Deploying env...")
-	env, err := opts.envDeployer.WaitForEnvironmentCreation(deployEnvInput)
-	if err != nil {
-		opts.prog.Stop("Error!")
-		return err
+	opts.prog.Start(fmt.Sprintf("Creating the infrastructure for the %s environment", color.HighlightUserInput(opts.EnvName)))
+	stackEvents, responses := opts.envDeployer.StreamEnvironmentCreation(deployEnvInput)
+	for stackEvent := range stackEvents {
+		opts.prog.Events(opts.humanizeEnvironmentEvents(stackEvent))
 	}
-	if err := opts.envCreator.CreateEnvironment(env); err != nil {
-		opts.prog.Stop("Error!")
-		return err
+	resp := <-responses
+	if resp.Err != nil {
+		opts.prog.Stop(fmt.Sprintf("%s Failed to create the infrastructure for the %s environment", color.ErrorMarker, color.HighlightUserInput(opts.EnvName)))
+		return resp.Err
+	}
+	opts.prog.Stop(fmt.Sprintf("%s Created the infrastructure for the %s environment", color.SuccessMarker, color.HighlightUserInput(opts.EnvName)))
+	if err := opts.envCreator.CreateEnvironment(resp.Env); err != nil {
+		return fmt.Errorf("store environment: %w", err)
 	}
-	opts.prog.Stop("Done!")
 	log.Successf("Created environment %s in region %s under project %s.\n",
-		color.HighlightUserInput(env.Name), color.HighlightResource(env.Region), color.HighlightResource(env.Project))
+		color.HighlightUserInput(resp.Env.Name), color.HighlightResource(resp.Env.Region), color.HighlightResource(resp.Env.Project))
 	return nil
 }
 
+func (opts *InitEnvOpts) humanizeEnvironmentEvents(resourceEvents []deploy.ResourceEvent) []termprogress.TabRow {
```

Comment:
I feel like this should live closer to the deploy package.

Reply:
The struggle is that the `textVPC`, ... constants I think should be in the `cli` package or some sort of view package.

I'll keep them here for now, let me know if you can think of a better place